build/buildConfig.js

```diff
@@ -139,7 +140,7 @@ module.exports = (bundles, themeEntries, paths, extractThemesPlugin, prod, publi
         }] : [])
     },
     devServer: {
-        proxy: {
```

Comment:
Another idea is to replace the last argument with a generic object that override all the previous configurations. This way webpack configuration is fully overridable and extensible. 
for instance, something named `webpackConfigs`
```
{
 devServer: {...},
  additionalPlugins: //<-- just for example, this can be future entry that can add webpack plugins to the existing ones. Without having an infinite argument list
 plugins: [...] // <-- Completely override webpackPlugins. 
..,and so on
}
```